src/tpm2.c

```diff
@@ -119,8 +119,10 @@ static TPM_RC TPM2_SendCommandAuth(TPM2_CTX* ctx, TPM2_Packet* packet,
         TPM2_Packet_ParseU32(packet, &authSz);
         /* Pass the Auth Session area, size field already passed by ParseU32 */
         packet->pos += authSz;
-        /* Extract the size of the first parameter */
-        TPM2_Packet_ParseU16(packet, &requestParamSz);
+        if (packet->pos + sizeof(requestParamSz) <= cmdSz) {
```

Comment:
If I understand it is valid to have no parameters and therefore have `requestParamSz=0` and this case want' handled previously?

Reply:
This was broken in #122 where the assumption was made that a request size was always present, which it is not.